tabular/src/autogluon/tabular/learner/abstract_learner.py

```diff
@@ -232,10 +254,15 @@ def score_debug(self, X: DataFrame, y=None, extra_info=False, compute_oracle=Fal
         for model_name, y_pred_proba_internal in model_pred_proba_dict.items():
             if self.eval_metric.needs_pred:
                 y_pred = self.label_cleaner.inverse_transform_proba(y_pred_proba_internal, as_pred=True)
+            else:
+                y_pred = self.label_cleaner.inverse_transform_proba(y_pred_proba_internal, as_pred=False)
+                y = y_internal
```

Comment:
bug: this is changing y in outer scope. Should use a temporary variable in here such as:

```
if ...:
    y_tmp = y
else:
    y_tmp = y_internal
scores = self.eval_metric (y_tmp, y_pred)
```

Reply:
good catch!